src/cascadia/TerminalApp/Command.h

```diff
@@ -0,0 +1,45 @@
+﻿/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- Command.h
+
+Abstract:
+- A command represents a single entry in the Command Palette. This is an object
+  that has a user facing "name" to display to the user, and an associated action
+  which can be dispatched.
+
+- For more information, see GH#2046, #5400, #5674, and #6635
+
+Author(s):
+- Mike Griese - June 2020
+
+--*/
+#pragma once
+
+#include "Command.g.h"
+#include "TerminalWarnings.h"
+#include "..\inc\cppwinrt_utils.h"
+
+namespace winrt::TerminalApp::implementation
+{
+    struct Command : CommandT<Command>
+    {
+        Command() = default;
+
+        static winrt::com_ptr<Command> FromJson(const Json::Value& json, std::vector<::TerminalApp::SettingsLoadWarnings>& warnings);
+        static std::vector<::TerminalApp::SettingsLoadWarnings> LayerJson(std::unordered_map<winrt::hstring, winrt::TerminalApp::Command>& commands,
+                                                                          const Json::Value& json);
+
+        WINRT_CALLBACK(PropertyChanged, Windows::UI::Xaml::Data::PropertyChangedEventHandler);
+        OBSERVABLE_GETSET_PROPERTY(winrt::hstring, Name, _PropertyChangedHandlers);
```

Comment:
FYI, `_UpperCase` tokens are reserved for the C++ language. consider alternative naming. see [this page](https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL51-CPP.+Do+not+declare+or+define+a+reserved+identifier#DCL51-CPP.Donotdeclareordefineareservedidentifier-NoncompliantCodeExample%28HeaderGuard%29) for details